### DISCUSSION TITLE
Is Codable still slower?

### DISCUSSION THREAD

**Discussion by Lancelotbronner** [airbnb/lottie-ios#2448] [General]
## Which Version of Lottie are you using?

Lottie 4.5.3

## Overview

New iOS versions have a new implementation for `JSONDecoder`/`JSONEncoder` that is multiple times faster than `JSONSerialization` and uses less memory.

When was the last benchmark made and on which devices? Where can we find those benchmarks?

**Comment by calda** [comment]
This was the PR that added the Dictionary-based decoding implementation: https://github.com/airbnb/lottie-ios/pull/1561

There hasn't been discussion about this recently. If somebody investigates and finds that Codable is now as fast or faster than the Dictionary approach, we can reassess our decoding approach.

**Reply by Lancelotbronner** [reply]
Doing the tests right now on my M1 (macOS 15 beta 3), Codable is 1.20x times slower for simple and 1.35x slower for complex.

I noticed there are a lot of classes in internals where structs seem a perfect fit, I'll try converting to structs.

**Reply by Lancelotbronner** [reply]
Just converting the layer models to structs brought it down to 1.14x and 1.15x. Complex took 8s.

**Reply by Lancelotbronner** [reply]
I'll have to leave the rest of the conversion to my backlog, I've got other priorities at the moment.
